include/ignition/math/AdditivelySeparableScalarField3.hh

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#ifndef IGNITION_MATH_SEPARABLE_SCALAR_FIELD3_HH_
+#define IGNITION_MATH_SEPARABLE_SCALAR_FIELD3_HH_
+
+#include <limits>
+#include <utility>
+
+#include <ignition/math/Region3.hh>
+#include <ignition/math/Vector3.hh>
+#include <ignition/math/config.hh>
+
+namespace ignition
+{
+  namespace math
+  {
+    // Inline bracket to help doxygen filtering.
+    inline namespace IGNITION_MATH_VERSION_NAMESPACE {
+    //
+    /** \class AdditivelySeparableScalarField3\
+     * AdditivelySeparableScalarField3.hh\
+     * ignition/math/AdditivelySeparableScalarField3.hh
+     */
+    /// \brief The AdditivelySeparableScalarField3 class constructs
+    /// a scalar field F in R^3 as a sum of scalar functions i.e.
+    /// F(x, y, z) = k (p(x) + q(y) + r(z)).
+    ///
+    /// ## Example
+    ///
+    /// \snippet examples/additively_separable_scalar_field3_example.cc complete
+    template<typename ScalarFunctionT, typename ScalarT>
```

Comment:
It would be good to document the `tparam`s here

Reply:
Done in https://github.com/ignitionrobotics/ign-math/pull/397/commits/b27d5f0af375be2b5d32685fbadf21117ef0517c.